### PR TITLE
Namespacing: Datadog::Core::Utils

### PR DESCRIPTION
**Pull request by delner**
This pull request moves `Utils` to `Datadog::Core::Utils`.

### Related PRs

It's part of a series of `Core` namespacing PRs derived from `refactor/core-namespacing`:

1. #1847
2. #1848
3. #1850 
4. #1851
5. **=> Namespacing: Datadog::Core::Utils**
6. #1853 
7. #1854 
8. #1855 
9. #1856 
10. #1857
11. #1858
12. #1859 
13. #1860 
14. #1861 
15. #1862

### CODE REVIEW

File: lib/ddtrace/sync_writer.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # typed: true
 require 'ddtrace/pipeline'
 require 'ddtrace/runtime/metrics'
-require 'ddtrace/utils/only_once'
+require 'datadog/core/utils/only_once'
 require 'ddtrace/writer'
```
Comment (ivoanjo) [inline lib/ddtrace/sync_writer.rb:5]: This can be removed -- this class no longer references `OnlyOnce`.

File: lib/ddtrace/tracer.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@
 require 'ddtrace/span_operation'
 require 'ddtrace/trace_flush'
 require 'ddtrace/trace_operation'
-require 'ddtrace/utils/only_once'
+require 'datadog/core/utils/only_once'
```
Comment (ivoanjo) [inline lib/ddtrace/tracer.rb:18]: This can be removed, no longer used by this class.

File: lib/ddtrace/writer.rb [hunk 3]
```diff
@@ -7,7 +7,7 @@
 require 'ddtrace/encoding'
 require 'ddtrace/workers'
 require 'ddtrace/diagnostics/environment_logger'
-require 'ddtrace/utils/only_once'
+require 'datadog/core/utils/only_once'
```
Comment (ivoanjo) [inline lib/ddtrace/writer.rb:10]: This one can be removed as well, no longer used.

File: spec/datadog/core/utils_spec.rb [hunk 4]
```diff
@@ -1,7 +1,10 @@
 # typed: false
+require 'spec_helper'
```
Comment (ivoanjo) [inline spec/datadog/core/utils_spec.rb:2]: Minor: This is not needed -- our `.rspec` already includes `--require spec_helper` which is automatically added when running RSpec.

**Review by ivoanjo — COMMENTED** [review]
Move looks good! But I'd like to avoid moving the profiling-specific classes into the proper core, as within weeks nobody will be using them.

File: lib/datadog/core/utils/database.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# typed: true
+module Datadog
+  module Core
+    module Utils
+      # Common database-related utility functions.
+      module Database
```
[the remark this answers is not in the document]
Comment (delner) [reply lib/datadog/core/utils/database.rb:6]: That seems reasonable.

File: lib/datadog/core/utils/object_set.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
+# typed: true
+require 'datadog/core/utils/sequence'
+
+module Datadog
+  module Core
+    module Utils
+      # Acts as a unique dictionary of objects
+      class ObjectSet
```
[the remark this answers is not in the document]
Comment (delner) [reply lib/datadog/core/utils/object_set.rb:8]: Regarding `ObjectSet`, `StringTable`, `Sequence`, we will likely use these in tracing in the near future for the purposes of transport, maybe even in the traces themselves to de-dup strings like tags. I'd prefer we keep these in `Core` for now. I don't want them deleted from profiling before we have a chance to use them.

**Review by marcotc — APPROVED** [review]

**Pull request merged**